### PR TITLE
Bepro 1336 bounty owner should be able to edit the tags and description when bounty is still in draft

### PR DESCRIPTION
**Pull request by MarcusviniciusLsantos**

### CODE REVIEW

File: components/bounty/issue-body.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  const [files, setFiles] = useState<IFilesProps[]>([]);
+  const [isPreview, setIsPreview] = useState<boolean>(false);
+  const [selectedTags, setSelectedTags] = useState<string[]>();
+  const [isUploading, setIsUploading] = useState<boolean>(false);
+  const { getDatabaseBounty } = useBounty();
+  const { state, dispatch } = useAppState();
+
+  const { updateIssue } = useApi();
+
+  useEffect(() => {
```
Comment (moshmage) [inline components/bounty/issue-body.tsx:43]: use named useEffects

File: pages/api/issue/[...ids].ts [hunk 2]
```diff
@@ -44,16 +53,82 @@ async function get(req: NextApiRequest, res: NextApiResponse) {
...
+    where: {
+      issueId,
+      network_id: network?.id
+    },
+    include: [{ association: "repository" }]
+  });
+
+  if (!issue) return res.status(404).json({message: "bounty not found"});
+
+  if(issue.state === 'draft'){
```
Comment (moshmage) [inline pages/api/issue/[...ids].ts:88]: missing logic for if user making PUT actually is owner of the bounty

**Review by moshmage — CHANGES_REQUESTED** [review]

File: pages/api/issue/[...ids].ts (hunk 2 above)
Comment (MarcusviniciusLsantos) [reply pages/api/issue/[...ids].ts:88]: the logic is in  `middleware\issue-route.ts`

File: components/bounty/issue-body.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+              )}
+              <IssueEditTag
+                isEdit={isEditIssue}
+                selectedTags={selectedTags}
+                setSelectedTags={setSelectedTags}
+                preview={isPreview}
+              />
+              <div className="container">
+                <IssueDescription
+                  body={isPreview ? addFilesInDescription(body) : body}
```
[the remark this answers is not in the document]
Comment (MarcusviniciusLsantos) [reply components/bounty/issue-body.tsx:122]: this logic is for when it is connected to the metamask and the other when it is not connected to the metamask

[the remark this answers is not in the document]
Comment (moshmage) [reply components/bounty/issue-body.tsx:122]: ?

**Pull request merged**